parsel/selector.py

```diff
@@ -222,33 +231,39 @@ def remove(self) -> None:
             x.remove()
 
 
+_NOTSET = object()
+
+
+def _load_json_or_none(text):
+    try:
+        return json.loads(text)
+    except ValueError:
+        return None
+
+
 class Selector:
     """
     :class:`Selector` allows you to select parts of an XML or HTML text using CSS
     or XPath expressions and extract data from it.
 
     ``text`` is a `str`` object
 
-    ``type`` defines the selector type, it can be ``"html"``, ``"xml"`` or ``None`` (default).
-    If ``type`` is ``None``, the selector defaults to ``"html"``.
+    ``type`` defines the selector type. It can be ``"html"`` (default),
+    ``"json"``, or ``"xml"``.
```

Comment:
"text" is also supported; is it intentional that we don't document it? I also wonder if there is a better terminology available than "selector type", though no suggestions :)

Reply:
I still don't get the benefit of the type as a user argument. The selector is a wrapper that has its logic figured out by which methods it uses and which data it finds per query. Allowing the user to set the type (besides purely for backward compatibility) seems to only allow it to break if this config was done poorly by the user. Without it, the Selector will either work or not depending purely on the methods used (unless I am missing out on something).

After saying that, I would say I don't find the `"text"` type worth documenting and if @Gallaecio idea of just letting it be `None` has no pending technical issues, then we should remove `"text"` altogether.

Reply:
After fiddling a bit with the code again, I believe `"text"` is needed, because [`None` needs to imply one of `"html"` or `"xml"`](https://github.com/EchoShoot/parsel/blob/812b2bf05b9c29827efb4a784cf37525f0e1132f/parsel/selector.py#L337) for backward compatibility. But I still would not document it for the reasons I explained in my earlier comment, i.e. is not meant for users to use.